processing/src/main/java/io/druid/segment/data/GenericIndexedWriter.java

```diff
@@ -82,6 +132,11 @@ public void write(T objectToWrite) throws IOException
     valuesOut.write(bytesToWrite);
 
     headerOut.write(Ints.toByteArray((int) valuesOut.getCount()));
+    headerOutLong.write(Longs.toByteArray(valuesOut.getCount()));
```

Comment:
For that I need to build headerOutLong from headerOut values during switch, I decided to keep the same info in headerOutLong to avoid building from headerOut and delete it if not required.